app/org/sagebionetworks/bridge/models/healthdata/HealthDataUserConsent.java

```diff
@@ -0,0 +1,85 @@
+package org.sagebionetworks.bridge.models.healthdata;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.joda.time.DateTime;
+
+import org.sagebionetworks.bridge.json.DateTimeToStringSerializer;
+import org.sagebionetworks.bridge.json.JodaDateTimeDeserializer;
+
+/**
+ * User consent metadata stored with the health data record. We store this in the Health Data Record table, so we can
+ * avoid doing a bunch of extra calls to other tables during export.
+ */
+@JsonDeserialize(builder = HealthDataUserConsent.Builder.class)
+public class HealthDataUserConsent {
```

Comment:
Agree to all of this, the only thing I would add is that for the purpose of export, I don't think you need to know anything about the consent except the user's sharing preference. And this record doesn't record the newly added sharing enumeration which specifies no sharing, share to researchers, or share to all researchers (see SharingScope). That's really all you need to know at the time of export, granted this enumeration may change making it more difficult.